server/service/src/sync/README.md

```diff
@@ -174,3 +172,10 @@ When change allows for previous version to still work without logical or syntax
 ### V6 Versioning
 
 The same versioning pattern also applies to the V6 sync (syncing with Open mSupply Central). The V6 sync version of the remote site is set in [settings.rs](./settings.rs), and checked in [sync_on_central](./sync_on_central/mod.rs)
+
+## Debugging sync::test::pull_and_push::test_sync_pull_and_push
+
+This test is a little tricky to debug when it fails, as the error messages are not very specific. While we're improving that, here are two strategies that can help isolate the root cause:
+
+1. Find the functions that may be being called by that test, and put in a println! statement to each one, then run the test again - that will show you which functions are called, how many times, and in what sequence.
+2. Search the error message displayed in the console for SQL errors. For example, 'FOREIGN KEY CONSTRAINT FAILED' and similar. If you find one, then the root cause of the trouble is almost certainly that a bit of your mock data fails to fulfil one of the constraints of the table you're testing. To check out the constraints, open the database in a database browser (we often use DBeaver) and generate the table's DDL (Data Description Language).
```

Comment:
```suggestion
2. Search the error message displayed in the console for SQL errors. For example, 'FOREIGN KEY CONSTRAINT FAILED' and similar. If you find one, this means some of the data our tests cases are inserting don't fulfil one of the constraints of the table you're testing. To check out the constraints, open the database in a database browser (we often use [DBeaver](https://dbeaver.io/), or [DB Browser for SQLite](https://sqlitebrowser.org/)) and generate the table's DDL (Data Description Language).
```

I don't like sounding too certain about these things. Not actually 100% if it's appropriate to have this kind of info in the readme but I think it could be useful!